src/tests/Register.test.js

```diff
@@ -1,11 +1,79 @@
 import React from 'react';
-import ShallowRenderer from 'react-test-renderer/shallow';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { render, fireEvent, screen, waitForElement, waitFor } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import Register from "../register/Register";
 
 
-test('renders Register', () => {
-    const renderer = new ShallowRenderer();
-    renderer.render(<Register />);
-    expect(renderer.getRenderOutput()).toMatchSnapshot()
-});
\ No newline at end of file
+// Mock successful registration
+const server = setupServer(
+    rest.post('http://127.0.0.1:5000/register', (req, res, ctx) => {
+        return res(ctx.json({ message: "User was created successfully. A confirmation email has been sent via email. After confirming your email you can login." }))
```

Comment:
Also verify the req is appropriate. You can do this by :

`expect(req.body).toBe(<whatever you expect>)`

Reply:
first of all, can you show me how to insert the req inside the mock POST of mew? I cannot find any article on that. We need that before we can expect to check req.body, I believe.

Reply:
Ok, so I tried inserting payload this way and the test still pass, so hopefully this is the right way (I'm just trying to be creative here, not sure if this is right)
<img width="684" alt="Screen Shot 2020-06-23 at 9 21 01 am" src="https://user-images.githubusercontent.com/29667122/85344468-1cd67e00-b533-11ea-9dd3-65d544161587.png">

But when I use the line you told me above 

> Also verify the req is appropriate. You can do this by :
      expect(req.body).toBe(<whatever you expect>)

That doesn't work (the test fail saying `req` is undefined.

<img width="625" alt="Screen Shot 2020-06-23 at 9 25 30 am" src="https://user-images.githubusercontent.com/29667122/85344606-7fc81500-b533-11ea-8ad9-59c5f8c85a10.png">

Reply:
I've pushed the latest code with modifications as per request changes plus payload and expect fail test in case you want to try it out. @meenakshi-dhanani

Reply:
tried to add global variable req but test still fails. But I can see the expected payload is there.

![Screen Shot 2020-06-23 at 9 34 07 am (2)](https://user-images.githubusercontent.com/29667122/85345046-bb171380-b534-11ea-8657-3e57342998a5.png)

Reply:
done.

<img width="1680" alt="Screen Shot 2020-06-23 at 9 48 33 am" src="https://user-images.githubusercontent.com/29667122/85345809-daaf3b80-b536-11ea-9035-467ebdd38d04.png">